demisto_sdk/commands/coverage_analyze/tools.py

```diff
@@ -0,0 +1,84 @@
+'''
+    This file contains functions that are related to the coverage reports but not used in the demisto-sdk source.
+'''
+import json
+from datetime import datetime, timedelta
+from typing import Optional
+
+import requests
+
+from demisto_sdk.commands.common.logger import logging_setup
+
+ONE_DAY = timedelta(days=1)
+LATEST_URL = 'https://storage.googleapis.com/marketplace-dist-dev/code-coverage-reports/coverage-min.json'
+HISTORY_URL = 'https://storage.googleapis.com/marketplace-dist-dev/code-coverage-reports/history/coverage-min/{date}.json'
+
+
+logger = logging_setup(2)
+
+
+def get_total_coverage(filename: Optional[str] = None, date: Optional[datetime] = None) -> float:
+    '''
+        Args:
+            filename:   The path to the coverage.json/coverage-min.json file.
+            date:       A datetime object.
+        Returns:
+            A float representing the total coverage that was found.
+                from file in case that filename was given.
+                from history bucket in case that date was given.
+                from latest bucket in any other case.
+            Or
+                0.0 if any errors were encountered.
+    '''
+    coverage_field = 'total_coverage'
+    assert not(filename and date), 'you should provide a filename or date not both.'
+    try:
+        if filename:
+            with open(filename, 'r') as report_file:
+                result = json.load(report_file)
+            try:
+                return result[coverage_field]
+            except KeyError:
+                return result['totals']['percent_covered']
+        elif date is None:
+            url = LATEST_URL
+        else:
+            url = HISTORY_URL.format(date=date.strftime('%Y-%m-%d'))
+
+        res = requests.get(url)
+        res.raise_for_status()
+        return res.json()[coverage_field]
+    except Exception as error:
+        logger.info(error)
+        return 0.0
+
+
+def yield_dates(start_date: datetime, end_date: datetime):
+    temp_time = start_date
+    while temp_time <= end_date:
+        yield temp_time
+        temp_time += ONE_DAY
+
+
+def create_coverage_graph(start_date: datetime, filename: str):
+    '''
+        Args:
+            start_date: The date to start collecting coverage information.
+            filename:   The path to the png file containing the coverage graph.
+        Creates:
+            A graoh in png format representing the coverage information per day.
+
+    '''
```

Comment:
since the sdk does not require the `matplotlib` package, add a note in the docstring that it is required by the function.